wandb/sdk/launch/agent/agent.py

```diff
@@ -602,6 +602,8 @@ def _thread_run_job(
             job_tracker.run = run
         while self._jobs_event.is_set():
             if self._check_run_finished(job_tracker, launch_spec):
+                # Ensure run is actually cancelled
+                job_tracker.run.cancel()
```

Comment:
Let's wrap this in a try? Don't want to error on already cancelled runs

Reply:
Or even better, introduce a new method to all SubmittedRun classes that tries to cancel, and warns if the run was still running

Reply:
Best would be to force this thread to wait until the run resource no longer exists before being allowed to exit.